capeinfra/pipeline/batch.py

```diff
@@ -0,0 +1,177 @@
+            srvc_policy_attach="arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role",
+            opts=ResourceOptions(parent=self),
+        )
+        # TODO: remove when adding real policy above, this simply gives full s3
+        # access (ISSUE #73)
+        aws.iam.RolePolicyAttachment(
+            f"{name}-instnc-s3svcroleatch",
+            role=self.instance_role.name,
+            policy_arn="arn:aws:iam::aws:policy/AmazonS3FullAccess",
+            opts=ResourceOptions(parent=self),
+        )
+        self.instance_role_profile = get_instance_profile(
+            f"{name}-instnc-rl",
+            self.instance_role,
+        )
+
+        self.security_group = aws.ec2.SecurityGroup(
+            f"{self.name}-scrtygrp",
+            # TODO: fine tune security group (ISSUE #77)
+            # Currently does not allow any inbound requests into an instance
+            # Allows all outbound requests unbounded
+            egress=[
+                {
+                    "from_port": 0,
+                    "to_port": 0,
+                    "protocol": "-1",
+                    "cidr_blocks": ["0.0.0.0/0"],
+                }
+            ],
+            opts=ResourceOptions(parent=self),
+        )
+
+        self.placement_group = aws.ec2.PlacementGroup(
+            f"{self.name}-plcmntgrp",
+            strategy=aws.ec2.PlacementStrategy.CLUSTER,
+            opts=ResourceOptions(parent=self),
+        )
+
+        # self.key_pair = aws.ec2.KeyPair(f"{self.name}-kypr")
+
+        self.compute_environment = aws.batch.ComputeEnvironment(
+            f"{self.name}-btch",
+            service_role=self.service_role.arn,
+            type="MANAGED",
+            compute_resources=aws.batch.ComputeEnvironmentComputeResourcesArgs(
+                type="EC2",
+                instance_role=self.instance_role_profile.arn,
+                # TODO: add EC2 key pair (ISSUE #77)
+                # I don't think this is necessarily required if we don't plan on
+                # SSHing into the machines (plus inbound requests should
+                # probably be blocked anyway)
+                # ec2_key_pair=self.key_pair.key_name,
+                image_id=image_id,
+                placement_group=self.placement_group.name,
+                security_group_ids=[self.security_group.id],
+                subnets=subnets,
+                **resources,
+            ),
+            opts=ResourceOptions(parent=self),
+        )
+
+        # TODO: figure out a good fair share policy, for now allow a job to
+        # request as much as they want (ISSUE #78)
+        # self.scheduling_policy = aws.batch.SchedulingPolicy(
+        #     f"{self.name}-schdl-plcy",
+        #     fair_share_policy={
+        #         ...
+        #     },
+        #     opts=ResourceOptions(parent=self.compute_environment),
+        # )
+
+        self.job_queue = aws.batch.JobQueue(
+            f"{self.name}-job-q",
```

Comment:
take it or leave it, i've been using the format `rawq` or `dapq` (without the final `-` in yours)